src/Chainweb/Pact/PactService.hs

```diff
@@ -856,16 +856,26 @@ execValidateBlock
     => BlockHeader
     -> PayloadData
     -> PactServiceM cas PayloadWithOutputs
-execValidateBlock currHeader plData =
-    -- TODO: are we actually validating the output hash here?
-    withBatch $ withCheckpointerRewind mb "execValidateBlock" $ \pdbenv -> do
-        !result <- playOneBlock currHeader plData pdbenv
-        return $! Save currHeader result
+execValidateBlock currHeader plData = do
```

Comment:
@emilypi is this the correct place for payload Validations? Do we have a canonical place for doing all payload validation checks?

Reply:
Payload validation is done here and `execNewBlock` which is a place we may want to have redundant checks

Reply:
@fosskers would we have to add `validateSilenceDates` (see below) to `execNewBlock`, too?